### PR TITLE
remove calls to java stream()

### PR DESCRIPTION
**Pull request by stephane-airbyte**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37539?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#37540** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37540?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#37539** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37539?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#37538** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37538?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#37537** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37537?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#37518** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37518?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/kotlin/io/airbyte/cdk/integrations/base/ssh/SshBastionContainer.kt [hunk 1]
```diff
@@ -144,13 +144,8 @@ class SshBastionContainer : AutoCloseable {
          */
         fun getInnerContainerAddress(container: Container<*>): ImmutablePair<String, Int> {
             return ImmutablePair.of(
-                container.containerInfo.networkSettings.networks.entries
-                    .stream()
-                    .findFirst()
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/kotlin/io/airbyte/cdk/integrations/base/ssh/SshBastionContainer.kt:149]: findFirst().get().value would throw an exception if the stream is empty. first() throws an exception if the stream is empty.

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumer.kt [hunk 2]
```diff
@@ -160,18 +159,11 @@ constructor(
...
-                .collect(
-                    Collectors.toMap(
-                        { streamDescriptor: StreamDescriptor -> streamDescriptor },
-                        { streamDescriptor: StreamDescriptor ->
-                            StreamSyncSummary(
-                                Optional.of(getRecordCounter(streamDescriptor).get()),
-                            )
-                        },
-                    ),
+            streamNames.associateWith { streamDescriptor: StreamDescriptor ->
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumer.kt:162]: ooh, TIL associateWith

(i'm immediately switching to `associate` [here](https://github.com/airbytehq/airbyte/pull/38121/files#diff-1cf39e4350409debcfd4aa5ebf18ac80a12d0ecce9c0cfd0d727b8ff32397361R133) but it's still cool!)

**Review by edgao — APPROVED** [review]
rubberstamping this :P I only reviewed the first 10ish files

**Review by gisripa — APPROVED** [review]

**Comment by stephane-airbyte** [comment]
### Merge activity

* **May 23, 1:53 PM EDT**: @stephane-airbyte started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/37539).

**Pull request merged**